src/proxy/topic/ReliableTopicProxy.ts

```diff
@@ -150,6 +151,61 @@ export class ReliableTopicProxy<E> extends BaseProxy implements ITopic<E> {
         }
     }
 
+    publishAll(messages: any[]): Promise<void> {
+        const reliableTopicMessages: ReliableTopicMessage[] = [];
+        assertNotNull(messages);
+
+        try {
+            for (const message of messages) {
+                const reliableTopicMessage = new ReliableTopicMessage();
+                reliableTopicMessage.payload = this.serializationService.toData(message);
+                reliableTopicMessage.publishTime = Long.fromNumber(Date.now());
+                reliableTopicMessage.publisherAddress = this.localAddress;
+                reliableTopicMessages.push(reliableTopicMessage);
+            }
+            switch (this.overloadPolicy) {
+                case TopicOverloadPolicy.ERROR:
+                    for (const reliableTopicMessage of reliableTopicMessages) {
+                        this.addWithError(reliableTopicMessage);
+                    }
+                    break;
+                case TopicOverloadPolicy.DISCARD_NEWEST:
+                    for (const reliableTopicMessage of reliableTopicMessages) {
+                        this.addOrDiscard(reliableTopicMessage);
+                    }
+                    break;
+                case TopicOverloadPolicy.DISCARD_OLDEST:
+                    for (const reliableTopicMessage of reliableTopicMessages) {
+                        this.addOrOverwrite(reliableTopicMessage);
+                    }
+                    break;
+                case TopicOverloadPolicy.BLOCK:
+                    for (const reliableTopicMessage of reliableTopicMessages) {
+                        this.addWithBackoff(reliableTopicMessage);
+                    }
+                    break;
+                default:
+                    throw new RangeError('Unknown overload policy');
+            }
+        } catch (e) {
+            if (e instanceof SchemaNotReplicatedError) {
+                return this.registerSchema(e.schema, e.clazz).then(() => this.publishAll(messages));
+            }
+            throw e;
```

Comment:
Java wraps the error:

`throw (RuntimeException) peel(e, null,
                    "Failed to publish messages: " + payload + " to topic:" + getName());`


This converts exceptions to RuntimeException basically. We can throw a HazelcastError IMO instead of that since RuntimeException does not exist in node.js. Let's say we throw HazelcastError, we should add it to the documentation of ITopic.

Reply:
Done.

Reply:
Can you add the HazelcastError that can be thrown from ReliableTopic to ITopic interface? You can say that HazelcastError may be thrown (you should give a link to HazelcastError) when the method is invoked from a reliable topic and the messages cannot be published.